server/src/utils.ts

```diff
@@ -155,6 +155,15 @@ class Utils {
 	public addRandomPostfix(message: string, length = 2) {
 		return `${message}-${this.randomStringGenerator(length)}`;
 	}
+
+	public isObjectWithProperty<Y extends PropertyKey>
```

Comment:
what about calling this `hasProperty` instead? 

Or, you could use optional chaining instead, and eliminate this function entirely.

```typescript
if(object?.someProperty){ ... }
```
The only caveat to this is booleans or empty strings would be falsey. But for those situations, you could just check for the default value.
```typescript
if(object?.someBool === true) 
//or
if(object?.someBool === false)

if(object?.someString?.length > 0) 
```

Reply:
It's named the same in the vscode extension. I just kept the same name.